Does not compile with C++ 17 compiler

The code does not compile with a compiler in C++ 17 mode, such as gcc/cpp 11.3.

```
lib/assembler_0.h:94:40: error: ISO C++17 does not allow ‘register’ storage class specifier [-Werror=register]
   94 |   void cover_row(register unsigned int r);
      |                                        ^
```

The `register` keyword was deprecated and finally removed in C++ 17.

--- a/src/lib/assembler_0.cpp
+++ b/src/lib/assembler_0.cpp
@@ -918,12 +918,12 @@ void assembler_0_c::uncover(unsigned int col) {
 /* remove all the columns from the matrix in which the given
  * row contains ones
  */
-void assembler_0_c::cover_row(register unsigned int r) {
+void assembler_0_c::cover_row(unsigned int r) {
   for (unsigned int j = right[r]; j != r; j = right[j])
     cover(colCount[j]);
 }
 
-bool assembler_0_c::try_cover_row(register unsigned int r, unsigned int * columns) {
+bool assembler_0_c::try_cover_row(unsigned int r, unsigned int * columns) {
 
   memset(columns, 0, varivoxelEnd * sizeof(unsigned int));
 
@@ -950,15 +950,15 @@ bool assembler_0_c::try_cover_row(register unsigned int r, unsigned int * column
   return true;
 }
 
-void assembler_0_c::uncover_row(register unsigned int r) {
+void assembler_0_c::uncover_row(unsigned int r) {
   for (unsigned int j = left[r]; j != r; j = left[j])
     uncover(colCount[j]);
 }
 
-void assembler_0_c::remove_row(register unsigned int r) {
-  register unsigned int j = r;
+void assembler_0_c::remove_row(unsigned int r) {
+  unsigned int j = r;
   do {
-    register unsigned int u, d;
+    unsigned int u, d;
 
     colCount[colCount[j]]--;
 
@@ -972,8 +972,8 @@ void assembler_0_c::remove_row(register unsigned int r) {
   } while (j != r);
 }
 
-void assembler_0_c::remove_column(register unsigned int c) {
-  register unsigned int j = c;
+void assembler_0_c::remove_column(unsigned int c) {
+  unsigned int j = c;
   do {
     right[left[j]] = right[j];
     left[right[j]] = left[j];
@@ -982,8 +982,8 @@ void assembler_0_c::remove_column(register unsigned int c) {
   } while (j != c);
 }
 
-void assembler_0_c::reinsert_row(register unsigned int r) {
-  register unsigned int j = r;
+void assembler_0_c::reinsert_row(unsigned int r) {
+  unsigned int j = r;
   do {
     up(down(j)) = j;
     down(up(j)) = j;
@@ -1309,7 +1309,7 @@ void assembler_0_c::iterativeMultiSearch(void) {
       unsigned int s = colCount[c];
 
       if (s) {
-        register unsigned int j = right[c];
+        unsigned int j = right[c];
 
         while (j) {
 
@@ -1332,7 +1332,7 @@ void assembler_0_c::iterativeMultiSearch(void) {
       // it doesn't cost a lot of time, so let's keep it in for the moment
       if (s) {
         unsigned int currentHoles = holes;
-        register unsigned int j = right[varivoxelEnd];
+        unsigned int j = right[varivoxelEnd];
 
         while (j != varivoxelEnd) {
           if (colCount[j] == 0) {

--- a/src/lib/assembler_0.h
+++ b/src/lib/assembler_0.h
@@ -91,22 +91,22 @@ private:
   /* 2 helper functions that cover and uncover one
    * selected row
    */
-  void cover_row(register unsigned int r);
-  void uncover_row(register unsigned int r);
+  void cover_row(unsigned int r);
+  void uncover_row(unsigned int r);
 
   /* same as cover row, but aborting
    * as soon as one of the columns does contain a zero
    * and then uncovering all that was already done
    */
-  bool try_cover_row(register unsigned int r, unsigned int * columns);
+  bool try_cover_row(unsigned int r, unsigned int * columns);
 
   /* these 2 functions remove and reinsert rows from the matrix
    * they only remove the given row
    */
-  void remove_row(register unsigned int r);
-  void reinsert_row(register unsigned int r);
+  void remove_row(unsigned int r);
+  void reinsert_row(unsigned int r);
 
-  void remove_column(register unsigned int c);
+  void remove_column(unsigned int c);
 
   /* this function gets called whenever an assembly was found
    * when a call-back is available it will call getAssembly to

--- a/src/lib/assembler_1.cpp
+++ b/src/lib/assembler_1.cpp
@@ -747,8 +747,8 @@ assembler_1_c::errState assembler_1_c::createMatrix(bool keepMirror, bool keepRo
   return errorsState;
 }
 
-void assembler_1_c::remove_column(register unsigned int c) {
-  register unsigned int j = c;
+void assembler_1_c::remove_column(unsigned int c) {
+  unsigned int j = c;
   do {
     right[left[j]] = right[j];
     left[right[j]] = left[j];
@@ -839,10 +839,10 @@ unsigned int assembler_1_c::clumpify(void) {
   return removed;
 }
 
-void assembler_1_c::remove_row(register unsigned int r) {
-  register unsigned int j = r;
+void assembler_1_c::remove_row(unsigned int r) {
+  unsigned int j = r;
   do {
-    register unsigned int u, d;
+    unsigned int u, d;
 
     colCount[colCount[j]] -= weight[j];
 

--- a/src/lib/assembler_1.h
+++ b/src/lib/assembler_1.h
@@ -116,8 +116,8 @@ private:
   bool column_condition_fulfillable(int col);
 //  void rec(unsigned int next_row);
   void iterative(void);
-  void remove_row(register unsigned int r);
-  void remove_column(register unsigned int c);
+  void remove_row(unsigned int r);
+  void remove_column(unsigned int c);
   unsigned int clumpify(void);
 
 

